File: editor/en/[404]Sum of Left Leaves.py
"""
Given the root of a binary tree, return the sum of all left leaves. 

 A leaf is a node with no children. A left leaf is a leaf that is the left 
child of another node. 

 
 Example 1: 
 
 
Input: root = [3,9,20,null,null,15,7]
Output: 24
Explanation: There are two left leaves in the binary tree, with values 9 and 15 
respectively.
 

 Example 2: 

 
Input: root = [1]
Output: 0
 

 
 Constraints: 

 
 The number of nodes in the tree is in the range [1, 1000]. 
 -1000 <= Node.val <= 1000 
 

 Related Topics Tree Depth-First Search Breadth-First Search Binary Tree 👍 5022
 👎 290

"""


# leetcode submit region begin(Prohibit modification and deletion)
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def sumOfLeftLeaves(self, root: Optional[TreeNode]) -> int:
        if root is None:
            return 0

        sum_ = 0

        if root.left is not None:
            if root.left.left is None and root.left.right is None:
                sum_ += root.left.val
            else:
                sum_ += self.sumOfLeftLeaves(root.left)

        sum_ += self.sumOfLeftLeaves(root.right)
        return sum_
# leetcode submit region end(Prohibit modification and deletion)

File: editor/en/leetcode/editor/en/[404]Sum of Left Leaves.java

//leetcode submit region begin(Prohibit modification and deletion)

class Solution {
    public int sumOfLeftLeaves(TreeNode root) {
        if (root == null) {
            return 0;
        }

        int result = 0;
        Queue<TreeNode> queue = new LinkedList<>();
        queue.offer(root);

        while (!queue.isEmpty()) {
            TreeNode current = queue.poll();

            if (current.left != null) {
                if (current.left.left == null && current.left.right == null) {
                    result += current.left.val;
                } else {
                    queue.offer(current.left);
                }
            }

            if (current.right != null) {
                queue.offer(current.right);
            }
        }

        return result;
    }
}
//leetcode submit region end(Prohibit modification and deletion)
